src/cluster_info.rs

```diff
@@ -1471,4 +1644,159 @@ mod tests {
             .unwrap();
         assert!(val.verify());
     }
+
+    #[test]
+    fn test_describe_data_plane() {
+        // no nodes
+        let (num_layers, _) = ClusterInfo::describe_data_plane(0, 200, 200, false);
+        assert_eq!(num_layers, 0);
+
+        // 1 node
+        let (num_layers, _) = ClusterInfo::describe_data_plane(1, 200, 200, false);
+        assert_eq!(num_layers, 1);
+
+        // 10 nodes with fanout of 2 and hood size of 2
+        let (num_layers, _) = ClusterInfo::describe_data_plane(10, 2, 2, false);
+        assert_eq!(num_layers, 5);
+
+        // fanout + 1 nodes with fanout of 2 and hood size of 2
+        let (num_layers, _) = ClusterInfo::describe_data_plane(3, 2, 2, false);
+        assert_eq!(num_layers, 2);
+
+        // 10 nodes with fanout of 4 and hood size of 2 while growing
+        let (num_layers, _) = ClusterInfo::describe_data_plane(10, 4, 2, true);
+        assert_eq!(num_layers, 3);
+
+        // A little more realistic
+        let (num_layers, _) = ClusterInfo::describe_data_plane(100, 10, 10, false);
+        assert_eq!(num_layers, 3);
+
+        // A little more realistic with odd numbers
+        let (num_layers, _) = ClusterInfo::describe_data_plane(103, 13, 13, false);
+        assert_eq!(num_layers, 3);
+
+        // larger
+        let (num_layers, layer_indices) = ClusterInfo::describe_data_plane(10_000, 10, 10, false);
+        assert_eq!(num_layers, 201);
+        // distances between index values should be the same since we aren't growing.
+        let capacity = 10 / 2 * 10;
+        assert_eq!(layer_indices[1], 10);
+        layer_indices[1..layer_indices.len()]
+            .chunks(2)
+            .for_each(|x| {
+                if x.len() == 2 {
+                    assert_eq!(x[1] - x[0], capacity);
+                }
+            });
+
+        // massive
+        let (num_layers, layer_indices) =
+            ClusterInfo::describe_data_plane(500_000, 200, 200, false);
+        let capacity = 200 / 2 * 200;
+        let cnt = 500_000 / capacity + 1;
+        assert_eq!(num_layers, cnt);
+        // distances between index values should be the same since we aren't growing.
+        assert_eq!(layer_indices[1], 200);
+        layer_indices[1..layer_indices.len()]
+            .chunks(2)
+            .for_each(|x| {
+                if x.len() == 2 {
+                    assert_eq!(x[1] - x[0], capacity);
+                }
+            });
+        let total_capacity: usize = *layer_indices.last().unwrap();
+        assert!(total_capacity >= 500_000);
+
+        // massive with growth
+        let (num_layers, _) = ClusterInfo::describe_data_plane(500_000, 200, 200, true);
+        assert_eq!(num_layers, 3);
+    }
+
+    #[test]
+    fn test_localize() {
+        // go for gold
+        let (_, layer_indices) = ClusterInfo::describe_data_plane(500_000, 200, 200, false);
```

Comment:
Is calling `describe_data_plane` helpful here? I'd have an easier time grokking `localize()` if you passed it hard-coded `layer_indices`. As-is, I'd need to go grok `describe_data_plane()` first, which is only documented as "calc num layers and their respective indices".

Reply:
Yeah constructing layer_indices is just a little tedious and since there's a function who's only job is to define what the layers look like, I figured it made sense to use it? 

I can document describe_data_plane a little better for sure. Basically given a node count, neighborhood size, and an initial fanout, it calculates how many layers there are and at what index each layer begins.

I'll update the doc for describe_data_plane, it's definitely a little lacking. I'll properly document the other functions as well.